kotlin/src/test/kotlin/com/linecorp/armeria/server/FlowAnnotatedServiceTest.kt

```diff
@@ -0,0 +1,278 @@
+        assertThat(res.status()).isEqualTo(HttpStatus.OK)
+        assertThat(res.contentUtf8()).isEqualTo("OK")
+    }
+
+    @Test
+    fun test_cancellation() {
+        try {
+            client.get("/flow/cancellation").aggregate().get()
+        } catch (e: Exception) {
+            // do nothing
+        }
+        Thread.sleep(3000L)
+        assertThat(cancelled.get()).isTrue
+    }
+
+    companion object {
+        @JvmField
+        @RegisterExtension
+        val server = object : ServerExtension() {
+            override fun configure(sb: ServerBuilder) {
+                @Suppress("unused")
+                sb.apply {
+                    annotatedService("/flow", object {
+                        @Get("/byte-streaming")
+                        @ProducesOctetStream
+                        suspend fun byteStreaming(): Flow<ByteArray> = flow {
+                            emit("hello".toByteArray())
+                            emit("world".toByteArray())
+                        }
+
+                        @Get("/json-string-streaming")
+                        @ProducesJsonSequences
+                        suspend fun jsonStreamingString(): Flow<String> = flow {
+                            emit("hello")
+                            emit("world")
+                        }
+
+                        @Get("/json-obj-streaming")
+                        @ProducesJsonSequences
+                        suspend fun jsonStreamingObj(): Flow<Member> = flow {
+                            emit(Member(name = "foo", age = 10))
+                            emit(Member(name = "bar", age = 20))
+                            emit(Member(name = "baz", age = 30))
+                        }
+
+                        @Get("/event-streaming")
+                        @ProducesEventStream
+                        suspend fun eventStreaming(): Flow<ServerSentEvent> = flow {
+                            emit(
+                                ServerSentEvent
+                                    .builder()
+                                    .id("1")
+                                    .event("MESSAGE_DELIVERED")
+                                    .data("{\"message_id\":1}")
+                                    .build()
+                            )
+                            emit(
+                                ServerSentEvent
+                                    .builder()
+                                    .id("2")
+                                    .event("FOLLOW_REQUEST")
+                                    .data("{\"user_id\":123}")
+                                    .build()
+                            )
+                        }
+
+                        @Blocking
+                        @Get("/blocking-annotation")
+                        @ProducesText
+                        suspend fun blockingAnnotation(): Flow<String> = flow {
+                            checkNotNull(ServiceRequestContext.currentOrNull())
```

Comment:
nit: We could just call `ServiceRequestContext.current()`

Reply:
Thanks, 😅 I did a simple thing harder 😓..